storage/gc_queue.go

```diff
@@ -288,17 +292,134 @@ func (gcq *gcQueue) process(now roachpb.Timestamp, repl *Replica,
 	return nil
 }
 
+// processSequenceCache iterates through the local sequence cache entries,
+// pushing the transactions (in cleanup mode) for those entries which appear
+// to be old enough. In case the transaction indicates that it's terminated,
+// the sequence cache keys are included in the result.
+func processSequenceCache(r *Replica, now, cutoff roachpb.Timestamp) []roachpb.GCRequest_GCKey {
+	snap := r.store.Engine().NewSnapshot()
+	defer snap.Close()
+
+	txns := make(map[string]*roachpb.Transaction)
+	idToKeys := make(map[string][]roachpb.GCRequest_GCKey)
+	r.sequence.Iterate(snap, func(key, id []byte, v roachpb.SequenceCacheEntry) {
+		if !cutoff.Less(v.Timestamp) {
+			idStr := string(id)
+			txns[idStr] = &roachpb.Transaction{ID: id, Key: v.Key, Status: roachpb.PENDING}
+			idToKeys[idStr] = append(idToKeys[idStr], roachpb.GCRequest_GCKey{Key: key})
+		}
+	})
+
+	var wg sync.WaitGroup
+	wg.Add(len(txns))
+	for _, txn := range txns {
+		// Check if the Txn is still alive. If this indicates that the Txn is
+		// aborted and old enough to guarantee that any running coordinator
+		// would have realized that the transaction wasn't running by means
+		// of a heartbeat, then we're free to remove the sequence cache entry.
+		// In the most likely case, there isn't even an entry (which will
+		// be apparent by a zero timestamp and nil last heartbeat).
+		go pushTxn(r, now, txn, roachpb.CLEANUP_TXN, &wg)
```

Comment:
Would it make sense to move `processSequenceCache` earlier in `process` so we could push transactions from the sequence cache in the same pass as other transactions?